core/src/test/java/org/apache/calcite/test/RelMetadataTest.java

```diff
@@ -273,6 +273,104 @@ final RelMetadataFixture sql(String sql) {
                 / (DEPT_SIZE + EMP_SIZE)));
   }
 
+  // ----------------------------------------------------------------------
+  // Tests for getForeignKeys
+  // ----------------------------------------------------------------------
+
+  @Test void testForeignKeysAggregateCallEmpty() {
+    sql("select hiredate, sum(sal) from emp group by hiredate")
+        .assertForeignKeysAreEmpty();
+  }
+
+  @Test void testForeignKeysAggregateKeyEmpty() {
+    sql("select hiredate, sum(sal) from emp group by hiredate")
+        .assertForeignKeysAreEmpty();
+  }
+
+  @Test void testForeignKeysAggregateKey() {
+    sql("select count(sal), deptno from emp group by deptno")
+        .assertForeignKeys(equalTo(bitSetOf(1)));
+  }
+
+  @Test void testForeignKeysTableOnly() {
+    sql("select deptno, ename from emp")
+        .assertForeignKeys(equalTo(bitSetOf(0)));
+  }
+
+  @Test void testForeignKeysTableOnlyEmpty() {
+    sql("select ename, job from emp")
+        .assertForeignKeysAreEmpty();
+  }
+
+  @Test void testForeignKeysExpressionEmpty() {
+    sql("select deptno + 1, ename from emp")
+        .assertForeignKeysAreEmpty();
+  }
+
+  @Test void testForeignKeysFilter() {
```

Comment:
Maybe add a test using the FK column in the filter too?

Reply:
Add testForeignKeysFilter and testForeignKeysFilterEmpty case.